### PR TITLE
wine 4.0

### PR DESCRIPTION
**Pull request by martell**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting? I can't on a newer macOS version
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/wine.rb [hunk 1]
```diff
@@ -34,8 +34,8 @@ class Wine < Formula
     url "https://source.winehq.org/git/wine.git"
 
     resource "mono" do
-      url "https://dl.winehq.org/wine/wine-mono/4.7.3/wine-mono-4.7.3.msi"
-      sha256 "d24a8017371c7e8224a1778bb43a113ed7ed9720efd9d0cda175d42db6106d3a"
+      url "https://dl.winehq.org/wine/wine-mono/4.7.5/wine-mono-4.7.5.msi"
```
Comment (SMillerDev) [inline Formula/wine.rb]: Since these seem to be the same, could you move them out of the `head do` block?

**Review by SMillerDev — CHANGES_REQUESTED** [review]
I was expecting this one, thank for the PR @martell . Just one comment for me.

**Review by SMillerDev — APPROVED** [review]

File: Formula/wine.rb [hunk 2]
```diff
@@ -9,21 +9,16 @@ class Wine < Formula
...
 
   stable do
-    url "https://dl.winehq.org/wine/source/3.0/wine-3.0.4.tar.xz"
-    mirror "https://downloads.sourceforge.net/project/wine/Source/wine-3.0.4.tar.xz"
-    sha256 "d45a88edbe7db363e297cacedc94e66df3464504a01c2eccf04f518066a6fb0c"
+    url "https://dl.winehq.org/wine/source/4.0/wine-4.0.tar.xz"
+    mirror "https://downloads.sourceforge.net/project/wine/Source/wine-4.0.tar.xz"
+    sha256 "6736cdee95b2b8bb021ec0c19497ed8cad5ae2c8bfdb7ab5dc687ff92a480d4d"
 
     # Patch to fix screen-flickering issues. Still relevant on 3.0.
```
Comment (fxcoudert) [inline Formula/wine.rb]: The patch should be removed.

File: Formula/wine.rb [hunk 3]
```diff
@@ -9,21 +9,16 @@ class Wine < Formula
...
     # https://bugs.winehq.org/show_bug.cgi?id=34166
     patch do
       url "https://raw.githubusercontent.com/Homebrew/formula-patches/74c2566/wine/2.14.patch"
       sha256 "6907471d18996ada60cc0cbc8462a1698e90720c0882846dfbfb163e5d3899b8"
     end
-
-    resource "mono" do
-      url "https://dl.winehq.org/wine/wine-mono/4.7.3/wine-mono-4.7.3.msi"
```
Comment (fxcoudert) [inline Formula/wine.rb]: According to https://source.winehq.org/git/wine.git/blob/refs/tags/wine-4.0:/dlls/appwiz.cpl/addons.c the correct Mono version for Wine 4.0 is 4.7.5. Please update the mono resource, and the URL at line 6 of the formula.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/wine.rb (hunk 3 above)
Comment (SMillerDev) [reply Formula/wine.rb]: https://github.com/Homebrew/homebrew-core/pull/36323/files#diff-129c0a4eb28c8c04cf7a1f694e1633e6R46 for the mono update

File: Formula/wine.rb (hunk 2 above)
Comment (martell) [reply Formula/wine.rb]: I had removed this locally previously but even though it was marked as a won't fix / resolved a long time ago it was still present on 3.0, I could update the patch to work on 4.0 if that is more suitable? @fxcoudert @SMillerDev 
https://bugs.winehq.org/show_bug.cgi?id=34166
Either way I will find time to do whichever choice is made tonight ( PST )

**Comment by fxcoudert** [comment]
Thanks @martell for the pull request, I am going to merge and fix the test later (the build is simply too long).

**Comment by fxcoudert** [comment]
Test fixed in 7efe23af56061b14ab3514d18f4ce9ebae7ebb1f

**Pull request closed without merging**